RUC LSM update

tanyasmirnova opened the pull request:
This PR includes:
1. Adding leaf area index, wilting point and soil field capacity to the history files.
2. Adding a placeholder for the fire heat flux provided by smoke-dust code.
3. A change to the transpiration function to take into account the non-linear distribution of roots.

tanyasmirnova committed:
Changes in RUC LSM:
1. adding a heat source from fires. It is icontrolled by a flag and not turned on yet. Needs more testing.
2. Adding output of surface paramteres: leaf area index (LAI), wilting point and soil field capacity.

tanyasmirnova committed:
Fire heat flux computation is added to the smoke-dust code.

tanyasmirnova committed:
Fix the segmentation fault error in computing fire heat flux.

grantfirl commented on physics/lsm_ruc.F90:
```diff
@@ -645,15 +649,38 @@ subroutine lsm_ruc_run                                            & ! inputs
-          !water and sea ice
-          smcref2 (i) = one
-          smcwlt2 (i) = zero
+        ! reassign smcref2 and smcwlt2 to RUC values at kdt=1
```
Would this be clearer to put the `if (kdt==1)` in the else statement for `if (.not. land(i))`? It looks like the same thing is done for the non-land points whether or not it is the first time step.

tanyasmirnova replied:
@grantfirl Agree, the change is made.

grantfirl commented on physics/module_sf_ruclsm.F90:
```diff
@@ -2049,6 +2068,9 @@ SUBROUTINE SFCTMP (debug_print, delt,ktau,conflx,i,j,         & !--- input varia
           eeta = eetas*(one-snowfrac) + eeta*snowfrac
           qfx = qfxs*(one-snowfrac) + qfx*snowfrac
           hfx = hfxs*(one-snowfrac) + hfx*snowfrac
+          !IF ( add_fire_heat_flux ) then ! JLS
```
Any reason to keep this commented-out section here and in line 2119?

tanyasmirnova replied:
@grantfirl The commented out section is removed.

grantfirl reviewed (approved):
Looks OK to me with a couple minor comments.

tanyasmirnova committed:
Addressed the reviewer's comments.

tanyasmirnova committed:
Changes in module_sf_ruclsm.F90:
1. Removed commented lines in fire feedback to RUC LSM.
2. The coordinates of a test point are passed to the RUC LSM subroutines to make debug printing more manageable.
3. Reverted change in the TRANSF subroutine to original version. The non-linear root distribution needs scientific evaluation in the retro runs.

tanyasmirnova committed:
Merge remote-tracking branch 'upstream/ufs/dev' into ruclsm_lai

tanyasmirnova committed:
Based on 1-D testing in ESM-SnowMIP a change to computation of snow thermal conductivity is made in module_sf_ruclsm.F90.

tanyasmirnova committed:
In module_sf_ruclsm.F90: clean-up comments, changed the constant for hard snow slab thermal conductivity.

Qingfu-Liu reviewed (commented):
There are several lines of the code (physics/module_sf_ruclsm.F90) are commented out (lines 4083, 4086, 4540, 4543, 5217, 5220, 5793 and 5796, !fact = 5._kind_phys and !fact = 2._kind_phys), those lines can be removed.

tanyasmirnova committed:
Removed requested by the reviewer commented lines.

tanyasmirnova commented:
@Qingfu-Liu The commented lines are removed.

tanyasmirnova committed:
Computation of surface fire heat for use in LSM to rrfs_smoke_wrapper.
Also, added the fraction of grid cell that is burned out.

tanyasmirnova committed:
Small changes to fire feedback to RUC LSM.
One more variable is added: fraction of grid cell burned by the fire.
This fraction is used to take into account fire's effect on surface albedo.
Also, added some debug prints.
These changes will change the results only when fire feedback is turned on. The default is .false.
